pkgs/by-name/or/oreo-cursors-plus/package.nix

```
@@ -0,0 +1,67 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  ruby,
+  inkscape,
+  xorg,
+  cursorsConf ? null, # If set to a string, overwrites contents of './cursors.conf'
+}:
+with lib;
+  stdenv.mkDerivation rec {
+    pname = "oreo-cursors-plus";
+    version = "unstable-2023-06-05";
+    src = fetchFromGitHub {
+      owner = "Souravgoswami";
+      repo = "oreo-cursors";
+      # At the time of writing, there are no version tags. The author will add them starting with the next version release.
+      # Using the latest commit instead.
+      rev = "9133204d60ca2c54be0df03b836968a1deac6b20";
+      hash = "sha256-6oTyOQK7mkr+jWYbPNBlJ4BpT815lNJvsJjzdTmj+68=";
+    };
+
+    nativeBuildInputs = [ ruby inkscape xorg.xcursorgen ];
+
+    # './cursors.conf' contains definitions of cursor variations to generate.
+    configurePhase =
+      if (cursorsConf != null)
+      then ''
+        echo "${cursorsConf}" > ./cursors.conf
+        echo "--- Start cursors.conf ---"
+        cat ./cursors.conf
+        echo "--- End cursors.conf ---"
+      ''
+      else "";
+
+    # The repo already contains the default cursors pre-generated in './dist'. Just copy these if './cursors.conf' is not overwritten.
+    # Otherwise firs remove all default variations and build.
+    buildPhase =
+      if (cursorsConf == null)
+      then ''
+        echo ./cursors.conf not overwritten
+        echo Using pre-built cursors
+      ''
+      else ''
+        echo ./cursors.conf was overwritten
+        echo Generating custom cursors
+
+        rm -r dist
+        rm -r src/oreo_*
+        HOME=$TMP
+        ruby generator/convert.rb
+        make build
+      '';
+
+    installPhase = ''
+      install -dm 755 $out/share/icons
+      cp -dr --no-preserve='ownership' ./dist/* $out/share/icons/
+    '';
```

Maybe something like this would be possible?

```suggestion
    installPhase = ''
      runHook preInstall
    
      mkdir -p $out/share
      mv ./dist $out/share/icons
      
      runHook postInstall
    '';
```